src/metpy/calc/kinematics.py

```diff
@@ -209,66 +345,82 @@ def total_deformation(u, v, dx, dy):
+        [..., Z, Y, X] ordering or specify *_dim keyword arguments.
+    u, v, w : `pint.Quantity` or `xarray.DataArray` or None
+        N-dimensional arrays with units of velocity representing the flow, with a component of
+        the wind in each dimension. For 1D advection, use 1 positional argument (with `dx` for
+        grid spacing and `x_dim` to specify axis if not the default of -1) or use 1 applicable
+        keyword argument (u, v, or w) for proper physical dimension (with corresponding `d*`
+        for grid spacing and `*_dim` to specify axis). For 2D/horizontal advection, use 2
+        positional arguments in order for u and v winds respectively (with `dx` and `dy` for
+        grid spacings and `x_dim` and `y_dim` keyword arguments to specify axes), or specify u
+        and v as keyword arguments (grid spacings and axes likewise). For 3D advection,
+        likewise use 3 positional arguments in order for u, v, and w winds respectively or
+        specify u, v, and w as keyword arguments (either way, with `dx`, `dy`, `dz` for grid
+        spacings and `x_dim`, `y_dim`, and `vertical_dim` for axes).
+    dx, dy, dz: `pint.Quantity` or None, optional
+        Grid spacing in applicable dimension(s). If using arrays, each array should have one
+        item less than the size of `scalar` along the applicable axis. If `scalar` is an
+        `xarray.DataArray`, these are automatically determined from its coordinates, and are
+        therefore optional. Required if `scalar` is a `pint.Quantity`. These are keyword-only
+        arguments.
+    x_dim, y_dim, vertical_dim: int or None, optional
+        Axis number in applicable dimension(s). Defaults to -1, -2, and -3 respectively for
+        (..., Z, Y, X) dimension ordering. If `scalar` is an `xarray.DataArray`, these are
+        automatically determined from its coordinates. These are keyword-only arguments.
 
     Returns
     -------
-    N-dimensional array
+    `pint.Quantity` or `xarray.DataArray`
         An N-dimensional array containing the advection at all grid points.
 
     """
-    # This allows passing in a list of wind components or an array.
-    wind = _stack(wind)
-
-    # If we have more than one component, we need to reverse the order along the first
-    # dimension so that the wind components line up with the
-    # order of the gradients from the ..., y, x ordered array.
-    if wind.ndim > scalar.ndim:
-        wind = wind[::-1]
-
-    # Gradient returns a list of derivatives along each dimension. We convert
-    # this to an array with dimension as the first index. Reverse the deltas to line up
-    # with the order of the dimensions.
-    grad = _stack(gradient(scalar, deltas=deltas[::-1]))
-
-    # Make them be at least 2D (handling the 1D case) so that we can do the
-    # multiply and sum below
-    grad, wind = np.atleast_2d(grad, wind)
-
-    return (-grad * wind).sum(axis=0)
+    return sum(
```

Comment:
I'll have to take a look with the dx/dy automatic arguments again. It would have helped if I wrote a test for it right away rather than saying "I'll get to it later."

The missing minus sign...silly error on my part.